Services/Block/classes/class.ilBlockGUI.php

```diff
@@ -33,7 +35,7 @@ abstract class ilBlockGUI
     public const PRES_SEC_LIST = 2;		// secondary list panel
     public const PRES_MAIN_LIST = 3;		// main standard list panel
     public const PRES_MAIN_TILE = 4;		// main standard list panel
-    private int $offset;
+    private int $offset = 0;
```

Comment:
I suggest to switch the approach if you have to modify all occurrences anyway. You should introduce a value object like `BlockPage` with a `PageSize` and a `PageOffset`. Some blocks might not support a pagination, so this should be `nullable`.

Reply:
OK, my fault. I thought you introduced this `$offset` property.